### PR TITLE
feat: add sample and test for getting an access token from an imperso…

### PR DESCRIPTION
**Pull request by irataxy**
…nated SA

## Description

Fixes b:302748677
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by arithmetic1728 — APPROVED** [review]

**Review by arithmetic1728 — CHANGES_REQUESTED** [review]
LGTM for snippets.py. However, the snippets_test.py failed due to:

```
google.auth.exceptions.RefreshError: ('Unable to acquire impersonated credentials', '{\n  "error": {\n    "code": 404,\n    "message": "Not found; Gaia id not found for email [sa@project.service.gserviceaccount.com](mailto:sa@project.service.gserviceaccount.com)",\n    "status": "NOT_FOUND"\n  }\n}\n')
```

**Comment by Sita04** [comment]
@GoogleCloudPlatform/python-samples-reviewers 
Kindly help set up a similar CL: 572924785 testing resource.

File: auth/cloud-client/snippets_test.py [hunk 1]
```diff
@@ -42,3 +42,11 @@ def test_explicit_compute_engine():
 
     with credentials_patch:
         snippets.explicit_compute_engine(project)
+
+
+def test_accesstoken_from_impersonated_credentials():
+    impersonated_service_account = "sa@project.service.gserviceaccount.com"
```
Comment (rsamborski) [inline auth/cloud-client/snippets_test.py:48]: Please use `auth-samples-testing@python-docs-samples-tests.iam.gserviceaccount.com`

**Review by rsamborski — APPROVED** [review]
Access will be granted per cl/573156138

File: auth/cloud-client/snippets_test.py (hunk 1 above)
Comment (irataxy) [reply auth/cloud-client/snippets_test.py]: Done

**Comment by arithmetic1728** [comment]
test is still failing for the same reason

**Comment by rsamborski** [comment]
> test is still failing for the same reason

Fixed with cl/573772247

**Review by arithmetic1728 — APPROVED** [review]

**Pull request merged**